pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,490 @@
+				return
+			}
+			if !reflect.DeepEqual(got.State, test.want.State) {
+				t.Errorf("State = %v, Want = %v", got.State, test.want.State)
+				return
+			}
+		})
+	}
+}
+
+//TestCommTo is function to test CommTo().
+func TestCommTo(t *testing.T) {
+	commChan := make(map[string]chan interface{})
+	testInterface := make(chan interface{}, 1)
+	commChan["test"] = testInterface
+	dtContext := &DTContext{
+		CommChan: commChan,
+	}
+	var returnValue interface{}
+	tests := []struct {
+		name       string
+		modulename string
+		content    interface{}
+		wantErr    error
+	}{
+		{
+			//Failure Case
+			name:       "ModuleNotPresent",
+			modulename: "ModuleA",
+			content:    nil,
+			wantErr:    errors.New("Not found chan to communicate"),
+		},
+		{
+			//Success Case
+			name:       "ModulePresent",
+			modulename: "test",
+			content:    dttype.MembershipUpdate{AddDevices: []dttype.Device{{ID: "DeviceA", Name: "Router", State: "unknown"}}},
+			wantErr:    nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := dtContext.CommTo(test.modulename, test.content)
+			if err == nil {
+				returnValue = <-testInterface
+			} else {
+				returnValue = nil
+			}
+			if !reflect.DeepEqual(err, test.wantErr) {
+				t.Errorf("Error in CommTo %v, Want %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(returnValue, test.content) {
+				t.Errorf("Got %v on channel, Want % v on Channel", returnValue, test.content)
+				return
+			}
+		})
+	}
+}
+
+//TestHeartBeat is function to test HeartBeat().
+func TestHeartBeat(t *testing.T) {
+	tests := []struct {
+		name       string
+		dtc        *DTContext
+		moduleName string
+		content    interface{}
+		wantError  error
+	}{
+		{
+			//Success Case
+			name: "PingTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleA",
+			content:    "ping",
+			wantError:  nil,
+		},
+		{
+			//Failure Case
+			name: "StopTest",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			moduleName: "ModuleB",
+			content:    "stop",
+			wantError:  errors.New("stop"),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dtc := &DTContext{
+				ModulesHealth: test.dtc.ModulesHealth,
+			}
+			if err := dtc.HeartBeat(test.moduleName, test.content); !reflect.DeepEqual(err, test.wantError) {
+				t.Errorf("DTContext.HeartBeat() error = %v, Want = %v", err, test.wantError)
+			}
+		})
+	}
+}
+
+//TestGetMutex is function to test GetMutex().
+func TestGetMutex(t *testing.T) {
+	dtc := &DTContext{
+		DeviceMutex: &sync.Map{},
+	}
+	var testMutex *sync.Mutex
+	dtc.DeviceMutex.Store("DeviceB", "")
+	dtc.DeviceMutex.Store("DeviceA", testMutex)
+	tests := []struct {
+		name     string
+		want     *sync.Mutex
+		wantBool bool
+		deviceID string
+	}{
+		{
+			//Failure Case-No device present
+			name:     "UnknownDevice",
+			wantBool: false,
+			deviceID: "DeviceC",
```

Comment:
Order in ascending order  deviceA then deviceB.... so that if we add test cases later, we can have deviceD and its in order